evalml/tests/component_tests/decomposer_tests/test_stl_decomposer.py

```diff
@@ -346,3 +346,47 @@ def test_stl_decomposer_doesnt_modify_target_index(
 
     y_new = stl.inverse_transform(y_t)
     pd.testing.assert_index_equal(y_new.index, original_y_index)
+
+
+@pytest.mark.parametrize("set_coverage", [True, False])
+def test_stl_decomposer_get_trend_prediction_intervals(
+    set_coverage,
+    generate_seasonal_data,
+):
+    coverage = [0.75, 0.85, 0.95] if set_coverage else None
+    period = 7
+    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
+        period=period,
+        freq_str="D",
+        set_time_index=True,
+    )
+    X_train = X[: 15 * period]
+    y_train = y[: 15 * period]
+
+    y_validate = y[15 * period :]
+
+    stl = STLDecomposer()
+    stl.fit(X_train, y_train)
+
+    def assert_pred_interval_coverage(pred_interval):
+        expected_coverage = [0.95] if coverage is None else coverage
+        for cover_value in expected_coverage:
+            for key in [f"{cover_value}_lower", f"{cover_value}_upper"]:
+                pd.testing.assert_index_equal(
+                    pred_interval[key].index,
+                    y_validate.index,
+                )
+
+    trend_pred_intervals = stl.get_trend_prediction_intervals(
+        y_validate,
+        coverage=coverage,
+    )
+    assert_pred_interval_coverage(trend_pred_intervals)
+
+    # Set y.index to be non-datetime
+    y_validate = y_validate.reset_index(drop=True)
+    trend_pred_intervals = stl.get_trend_prediction_intervals(
```

Comment:
You should just parametrize this with a descriptive name for the parameter like "index_type": ["integer", "datetime"].  We should try to get away from single tests that have multiple, sequential assertions in them as we'll have to sometimes fix the first assertion, rerun the test, fix the second assertion, rerun the test...rather than just have the test failure parameters show us exactly what behavior we broke.